lib/autoPipelining.ts

```diff
@@ -104,7 +105,14 @@ export function executeWithAutoPipelining(
   }
 
   // If we have slot information, we can improve routing by grouping slots served by the same subset of nodes
-  const slotKey = client.isCluster ? client.slots[calculateSlot(args[0])].join(",") : 'main';
+  // Note that the first value in args may be a (possibly empty) array.
+  // ioredis will only flatten one level of the array, in the Command constructor.
+  const prefix = client.options.keyPrefix || "";
+  const slotKey = client.isCluster
+    ? client.slots[
+        calculateSlot(`${client.options.keyPrefix}${flatten(args)[0]}`)
```

Comment:
```suggestion
        calculateSlot(`${prefix}${flatten(args)[0]}`)
```

re: flatten - while it might be convenient, this is definitely a performance hit as it will be done on every operation

Reply:
Added a new function getFirstValueInFlattenedArray